pkg/reconciler/kubernetes/tektonhub/installerset.go

```diff
@@ -41,21 +41,32 @@ func (r *Reconciler) checkIfInstallerSetExist(ctx context.Context, oc clientset.
 			return false, err
 		}
 
+		var deleteInstallerSets bool
+
+		existingNamespace, ok := ctIs.Annotations[v1alpha1.TargetNamespaceKey]
+		if ok && existingNamespace != targetNamespace {
+			deleteInstallerSets = true
+		}
+
 		version, ok := ctIs.Annotations[v1alpha1.ReleaseVersionKey]
+		if ok && version != relVersion {
+			// release version doesn't exist or is different from expected
+			// mark the deleteInstallerSets true
+			deleteInstallerSets = true
+		}
```

Comment:
May be we can club this as we do for other components https://github.com/tektoncd/operator/blob/main/pkg/reconciler/kubernetes/tektonchain/tektonchain.go#L236-L244 and we might not even need that flag `deleteInstallerSets`